fv3gfs/physics/physics_state.py

```diff
@@ -1,75 +1,134 @@
-from fv3gfs.physics.stencils.microphysics import MicrophysicsState
-from fv3core.decorators import ArgSpec, get_namespace
-from fv3gfs.util.quantity import Quantity
-import fv3core.utils.gt4py_utils as utils
+import dataclasses
+from fv3core.utils.typing import FloatField, FloatFieldIJ
 import copy
 
+from fv3gfs.physics.stencils.microphysics import MicrophysicsState
+
 
+@dataclasses.dataclass()
 class PhysicsState:
-    arg_specs = (
-        ArgSpec("qvapor", "specific_humidity", "kg/kg", intent="inout"),
-        ArgSpec("qliquid", "cloud_water_mixing_ratio", "kg/kg", intent="inout"),
-        ArgSpec("qrain", "rain_mixing_ratio", "kg/kg", intent="inout"),
-        ArgSpec("qsnow", "snow_mixing_ratio", "kg/kg", intent="inout"),
-        ArgSpec("qice", "cloud_ice_mixing_ratio", "kg/kg", intent="inout"),
-        ArgSpec("qgraupel", "graupel_mixing_ratio", "kg/kg", intent="inout"),
-        ArgSpec("qo3mr", "ozone_mixing_ratio", "kg/kg", intent="inout"),
-        ArgSpec("qsgs_tke", "turbulent_kinetic_energy", "m**2/s**2", intent="inout"),
-        ArgSpec("qcld", "cloud_fraction", "", intent="inout"),
-        ArgSpec("pt", "air_temperature", "degK", intent="inout"),
-        ArgSpec(
-            "delp", "pressure_thickness_of_atmospheric_layer", "Pa", intent="inout"
-        ),
-        ArgSpec("delz", "vertical_thickness_of_atmospheric_layer", "m", intent="inout"),
-        ArgSpec("ua", "eastward_wind", "m/s", intent="inout"),
-        ArgSpec("va", "northward_wind", "m/s", intent="inout"),
-        ArgSpec("w", "vertical_wind", "m/s", intent="inout"),
-        ArgSpec("omga", "vertical_pressure_velocity", "Pa/s", intent="inout"),
-    )
+    """
+    Physics state variables
```

Comment:
We may want our docstrings to follow the google standard, so that Napoleon can parse them correctly when rendering them to html in sphinx. That was the motivation for putting it in the style guide, though since we don't currently have sphinx docs there are other places like this where we go freestyle.